pkg/exemplars/proxy.go

```diff
@@ -164,6 +158,15 @@ func (s *Proxy) Exemplars(req *exemplarspb.ExemplarsRequest, srv exemplarspb.Exe
 	return nil
 }
 
+func isExternalLabel(name string, sets []labels.Labels) bool {
```

Comment:
The function seems to be misnamed, there is nothing specific to external labels in the implementation.

Reply:
The function is checking if the label named `name` is present in the label set of the external labels announced by the ThanosStore (e.g. receiver). At the end the proxy is responsible for checking the external labels because the prometheus code for exemplars matching does not handle them. 

Feel free to suggest a better name, I could make it more generic and call `isPresentInLabelSet`.

Reply:
`containsLabelName` seems to be more appropriate since we can apply this to internal labels too.